src/store.rs

```diff
@@ -890,16 +890,14 @@ impl<F: LurkField> Store<F> {
     }
 
     pub fn open_mut(&mut self, ptr: Ptr<F>) -> Result<Ptr<F>, LurkError> {
-        assert!(ptr.0 == Tag::Comm || ptr.0 == Tag::Num);
-
         let p = match ptr.0 {
             Tag::Comm => ptr,
             Tag::Num => {
                 let scalar = self.fetch_num(&ptr).map(|x| x.into_scalar()).unwrap();
 
                 self.intern_maybe_opaque_comm(scalar)
             }
-            _ => unreachable!(),
+            _ => return Err(LurkError::Store("wrong type for hidden value tag".into())),
```

Comment:
This message is misleading. The hidden value is the one committed to, not the commitment specifier.

Maybe "wrong type for commitment specifier".